internal/commands/app/diff.go

```diff
@@ -187,3 +216,18 @@ func (i *diffInputs) Resolve(profile *user.Profile, ui terminal.UI) error {
 
 	return nil
 }
+
+func (i *diffInputs) resolveAppDependencies(rootDir string) (local.Dependencies, error) {
+	var err error
+	var appDependencies local.Dependencies
+	if i.IncludePackageJSON {
+		appDependencies, err = local.FindPackageJSON(rootDir)
```

Comment:
I would also vote that `package.json` takes precedences of any `node_modules` archive

it would make iterative development much easier (say if you needed to test out various versions of a library, this would just be simple JSON updates, rather than `yarn install` + some zip/tar command

Reply:
I'm ok with 

a) throwing an error and asking them to use one flag which goes in-line with our functionality. If someone is trying to use both, they should be explicitly told they can't. 

a) if both flags are specified, tell the user `Packages will be uploaded from package.json. Please exclusively use the `--include-dependencies` flag  to upload the dependencies archive`

Reply:
I will go with the first one then.